controllers/etcdcluster_controller.go

```diff
@@ -83,14 +86,130 @@ func (r *EtcdClusterReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error)
 			log.Error(err, "unable to create Service", "namespace", service.Namespace, "cluster", cluster.Name)
 			return ctrl.Result{}, err
 		}
+		log.V(1).Info("Created Service", "service name", service.Name)
 		return ctrl.Result{}, nil
 	}
+	log.V(2).Info("Service exists", "service name", service.Name)
+
+	peers := &etcdv1alpha1.EtcdPeerList{}
+	if err := r.List(ctx, peers, client.MatchingFields{"spec.clusterName": cluster.Name}); err != nil {
+		log.Error(err, "unable to list peers", "namespace", service.Namespace, "cluster", cluster.Name)
+		return ctrl.Result{}, err
+	}
+
+	// Assume we can't communicate with the cluster
+	log.V(1).Info("Unable to communicate with etcd cluster")
+	if int32(len(peers.Items)) < *cluster.Spec.Replicas {
+
+		// Create more peers
+		peerName := nextAvailablePeerName(cluster, peers.Items)
+		log.V(1).Info("Insufficient peers for replicas, adding new peer",
+			"Current number of peers", len(peers.Items),
+			"Desired number of replicas", cluster.Spec.Replicas,
+			"New peer's name", peerName)
```

Comment:
Keys in these structured logs should not contain whitespace or punctuation: https://github.com/go-logr/logr/blob/a1ebd699b1950beb1da0752cbb5559662018f798/logr.go#L85-L96

Reply:
Looking at it I'm also duplicating things as we already put the namespaced name on all the logs. I've gone through and improved the use of the structured logging.